types/fluent-react/index.d.ts

```diff
@@ -0,0 +1,106 @@
+}
+export interface LocalizationProviderProps {
+  bundles: IterableIterator<FluentBundle>;
+  parseMarkup?: MarkupParser;
+}
+export class LocalizationProvider extends React.Component<LocalizationProviderProps> {
+}
+
+export class ReactLocalization {
+  constructor(bundles: IterableIterator<FluentBundle>);
+  getString(id: string, args?: object, fallback?: string): string;
+}
+
+export interface LocalizedProps {
+  id: string;
+  attrs?: object;
+  [key: string]: any;
+}
+
+export class Localized extends React.Component<LocalizedProps> {
+}
+
+// Inspired by react-redux's type definition:
+/**
+ * A property P will be present if:
+ * - it is present in DecorationTargetProps
+ *
+ * Its value will be dependent on the following conditions
+ * - if property P is present in InjectedProps and its definition extends the definition
+ *   in DecorationTargetProps, then its definition will be that of DecorationTargetProps[P]
+ * - if property P is not present in InjectedProps then its definition will be that of
+ *   DecorationTargetProps[P]
+ * - if property P is present in InjectedProps but does not extend the
+ *   DecorationTargetProps[P] definition, its definition will be that of InjectedProps[P]
+ */
+export type Matching<InjectedProps, DecorationTargetProps> = {
+	[P in keyof DecorationTargetProps]: P extends keyof InjectedProps
+		? InjectedProps[P] extends DecorationTargetProps[P]
+			? DecorationTargetProps[P]
+			: InjectedProps[P]
+		: DecorationTargetProps[P];
+};
+
+/**
+ * a property P will be present if :
+ * - it is present in both DecorationTargetProps and InjectedProps
+ * - InjectedProps[P] can satisfy DecorationTargetProps[P]
+ * ie: decorated component can accept more types than decorator is injecting
+ *
+ * For decoration, inject props or ownProps are all optionally
+ * required by the decorated (right hand side) component.
+ * But any property required by the decorated component must be satisfied by the injected property.
+ */
+export type Shared<
+    InjectedProps,
+    DecorationTargetProps extends Shared<InjectedProps, DecorationTargetProps>
+    > = {
+        [P in Extract<keyof InjectedProps, keyof DecorationTargetProps>]?: InjectedProps[P] extends DecorationTargetProps[P] ? DecorationTargetProps[P] : never;
+    };
+
+// Infers prop type from component C
+export type GetProps<C> = C extends React.ComponentType<infer P> ? P : never;
+
+export type GetString = (id: string, args?: object) => string;
+
+export interface InjectedProps {
+  getString: GetString;
+}
+
+// Taken from
+// https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#predefined-conditional-types
+export type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
+
+// Injects `getString` and removes it from the prop requirements.
+// Will not pass through `getString` if it's passed in during
+// render.
+export interface WithLocalization {
+  <TNeedProps, C extends React.ComponentType<Matching<InjectedProps, GetProps<C>>>>(
```

Comment:
The `TNeedProps` type argument here isn't actually doing anything. You can see this in the test you provided since it shows that it's instantiated as `{}`, thereby not affecting the intersection where it's used.